src/plots/cartesian/axes.js

```diff
@@ -657,8 +657,35 @@ function isClose(a, b) {
 axes.prepTicks = function(ax, opts) {
     var rng = Lib.simpleMap(ax.range, ax.r2l, undefined, undefined, opts);
 
-    // calculate max number of (auto) ticks to display based on plot size
-    if(ax.tickmode === 'auto' || !ax.dtick) {
+    // sync ticks with the overlaying defined axis
+    if(ax.tickmode === 'sync') {
+        var baseAxis = ax._mainAxis;
+
+        // get range min and max to find range delta of axis 1
+        var minValBaseAxis = Math.min(baseAxis.range[0], baseAxis.range[1]);
+        var maxValBaseAxis = Math.max(baseAxis.range[0], baseAxis.range[1]);
+        var rangeDeltaBaseAxis = Math.abs(maxValBaseAxis - minValBaseAxis);
```

Comment:
Do we really need `Math.abs` here?
```suggestion 
        var rangeDeltaBaseAxis = maxValBaseAxis - minValBaseAxis;
```